test for unlimited string literal in IR via yul

clonker opened the pull request:
- Defines a contract with a name that exceeds 32 characters
- programSize is defined as datasize(contractName)
- if not correct, programSize can be substituted by optimizations into ctor return value
- this is due to u256 words only capturing first 32 chars

cameel commented on test/libyul/yulOptimizerTests/commonSubexpressionEliminator/long_literals_as_builtin_args.yul:
```diff
@@ -0,0 +1,32 @@
+        codecopy(i, dataoffset("AccessControlDefaultAdminRules4233_14_deployed"), j)
+    }
+    object "AccessControlDefaultAdminRules4233_14_deployed" {
+        code {
+            {
+                revert(0, 0)
+            }
+        }
```
You don't even need any code here for the example to make sense:
```suggestion
        code {}
```

nikola-matic replied:
`soltest` is failing because CSE (or whatever step is run in the yul optimizer test) will get rid of this anyway. Just leave the test as is, but update the expectations properly.

clonker committed:
test substitution of unlimited string literal in yul

- Defines a contract with a name that exceeds 32 characters
- programSize is defined as datasize(contractName)
- if not correct, programSize can be substituted by optimizations into ctor return value
- this is due to u256 words only capturing first 32 chars

nikola-matic reviewed (approved):
If the build passes (sans `ext_benchmarks` of course), feel free to merge. Or can you actually merge by yourself?

clonker commented:
> If the build passes (sans `ext_benchmarks` of course), feel free to merge. Or can you actually merge by yourself?

Can!
